srcpkgs/openttd/template

```diff
@@ -15,46 +15,67 @@ configure_args="
  -DCMAKE_INSTALL_BINDIR=bin
  -DCMAKE_INSTALL_DATADIR=/usr/share
  -DCMAKE_INSTALL_DOCDIR=share/doc/openttd
+ $(vopt_bool dedicated OPTION_DEDICATED)
 "
 
+build_options="gfx sfx msx dedicated"
+build_options_default="gfx sfx msx"
+desc_option_gfx="Include OpenGFX graphics pack"
+desc_option_sfx="Include OpenSFX sounds pack"
+desc_option_msx="Include OpenMSX music pack"
+desc_option_dedicated="Build dedicated server only, without GUI"
+
 hostmakedepends="pkg-config unzip tar"
-makedepends="SDL2-devel libcurl-devel freetype-devel fontconfig-devel
- harfbuzz-devel icu-devel libpng-devel liblzma-devel fluidsynth-devel"
-depends="hicolor-icon-theme fluidsynth soundfont-fluid"
+makedepends="$(vopt_if dedicated '' 'SDL2-devel freetype-devel fontconfig-devel
+ harfbuzz-devel icu-devel libpng-devel fluidsynth-devel') libcurl-devel
+ liblzma-devel"
+depends="$(vopt_if dedicated '' 'hicolor-icon-theme fluidsynth
+ soundfont-fluid')"
 short_desc="Open Source version of Transport Tycoon Deluxe"
 maintainer="Emil Tomczyk <emru@emru.xyz>"
 license="GPL-2.0-only, Zlib"
 homepage="https://www.openttd.org/"
 distfiles="https://cdn.openttd.org/openttd-releases/${version}/openttd-${version}-source.tar.xz
- https://cdn.openttd.org/opengfx-releases/${_gfxver}/opengfx-${_gfxver}-all.zip
- https://cdn.openttd.org/opensfx-releases/${_sfxver}/opensfx-${_sfxver}-all.zip
- https://cdn.openttd.org/openmsx-releases/${_msxver}/openmsx-${_msxver}-all.zip"
+ $(vopt_if gfx https://cdn.openttd.org/opengfx-releases/${_gfxver}/opengfx-${_gfxver}-all.zip)
```

Comment:
I don't think I like conditional distfiles; they will break helpers like `xgensum`.

Reply:
I was also thinking about moving resource packs to separate packages, many other distros do it. Would it be better in this case?

Reply:
Given that they are entirely separate distfiles, breaking them out as separate packages seems like a reasonable option. That is a much better experience for people who might wish to choose their own resources than forcing them to build a custom package.

Reply:
Then I'll do so, leaving here option for building only dedicated server